Penteract/Assets/Scripts/Fang.cpp

```diff
@@ -217,16 +268,21 @@ void Fang::PlayAnimation() {
 
 void Fang::Update(bool lockMovement, bool lockOrientation) {
 	if (isAlive) {
-		Player::Update(dashing, dashing);
-		if (Input::GetMouseButtonDown(2)) {
-			InitDash();
-		}
-		if (!dashing) {
-			if (Input::GetMouseButtonDown(0)) Shoot();
-		}
-		Dash();
-
-	} else {
+		if (EMP) {
+			Player::Update(dashing || EMP->IsActive(), dashing || EMP->IsActive());
+			if (Input::GetMouseButtonDown(2)) {
```

Comment:
You're not checking if EMP is currently being used, if you spam the dash button while EMP is happening, you get visual bugs on the HUD due to cooldown effects